src/__tests__/generations.servertest.ts

```diff
@@ -309,7 +309,7 @@ describe("/api/public/generations API Endpoint", () => {
     expect(dbGeneration2?.parentObservationId).toBe(generationId);
   });
 
-  it("should create trace when creating generation without existing trace with externalId", async () => {
+  it("should not create trace when creating generation without existing trace with externalId", async () => {
```

Comment:
No changer in behavior here. In the previous version, we did the following for observations

```
const traceId = !obj.traceId
        ? // Create trace if no traceid - backwards compatibility
          (
            await prisma.trace.create({
              data: {
                projectId: authCheck.scope.projectId,
                name: obj.name,
              },
            })
          ).id
        : obj.traceIdType === "EXTERNAL"
        ? // Find or create trace if externalTraceId
          (
            await prisma.trace.upsert({
              where: {
                projectId_externalId: {
                  projectId: authCheck.scope.projectId,
                  externalId: obj.traceId,
                },
              },
              create: {
                projectId: authCheck.scope.projectId,
                externalId: obj.traceId,
              },
              update: {},
            })
          ).id
        : obj.traceId;
```
In this case we set the traceId + insert EXTERNAL as type. This would upsert the trace. Otherwise, we only create the trace if the id is not provided. I doubt this even happens with our current SDK implementations.